test/e2e/network/service.go

```diff
@@ -615,6 +614,56 @@ func testHTTPHealthCheckNodePort(ip string, port int, request string) (bool, err
 	return false, fmt.Errorf("unexpected HTTP response code %s from health check responder at %s", resp.Status, url)
 }
 
+type healthzResponse struct {
+	Service struct {
+		Namespace string `json:"namespace"`
+		Name      string `json:"name"`
+	} `json:"service"`
+	LocalEndpoints int `json:"localEndpoints"`
+}
+
+func testHTTPHealthCheckNodePortFromTestContainer(config *e2enetwork.NetworkingTestConfig, host string, port int, timeout time.Duration, expectSucceed bool, threshold int) error {
+	count := 0
+	pollFn := func() (bool, error) {
+		rawResponse, err := config.GetResponseFromTestContainer(
+			"http",
+			"healthz",
+			host,
+			port)
+		if err != nil {
+			return false, nil
+		}
+		framework.Logf("Got response from health check node port via test container: %#v", rawResponse)
+		var success bool
+		if len(rawResponse.Responses) > 0 {
+			// We cannot get the status code if we curl the healthz endpoint in the test container,
```

Comment:
Is this really true? Cannot we introduce another method `GetHTTPCodeFromTestContainer` that would be executing something like `curl -w "%{http_code}" -s -o /dev/null <address>`? It'd simplify this method a lot